app/src/main/java/com/concordium/wallet/ui/common/TransactionViewHelper.kt

```diff
@@ -62,66 +67,130 @@ object TransactionViewHelper {
-            costTextView.visibility = View.VISIBLE
-            amountTextView.visibility = View.GONE
-            costTextView.text =
-                costTextView.context.getString(R.string.account_details_shielded_transaction_fee)
+            amountTextView.visibility = View.VISIBLE
+            amountTextView.text =
+                amountTextView.context.getString(R.string.account_details_shielded_transaction_fee)
         }
 
         fun showDecryptedValueOfEncryptedAmount() {
             //show decrypted value of encrypted amount
             lockImageView.visibility = View.VISIBLE
             totalTextView.visibility = View.GONE
-            costTextView.visibility = View.GONE
             amountTextView.visibility = View.GONE
             //...for now
         }
 
         fun hideCostLine() {
-            costTextView.visibility = View.GONE
             amountTextView.visibility = View.GONE
         }
 
 
+        @SuppressLint("SetTextI18n")
         fun showCostLineWithAmounts() {
             // Subtotal and cost
             if (ta.subtotal != null && ta.cost != null) {
-                costTextView.visibility = View.VISIBLE
                 amountTextView.visibility = View.VISIBLE
 
                 var cost = ta.cost
                 var costPrefix = ""
+                var amountColorSpan: SpannableString
+                var costColorSpan: SpannableString
+
+                val textBuilder = SpannableStringBuilder()
+                val amountText =
+                    "${CurrencyUtil.formatGTU(ta.subtotal, withGStroke = true)} - "
+                val costText by lazy { "$costPrefix${
+                    CurrencyUtil.formatGTU(
+                        cost, withGStroke = true
+                    )
+                } ${amountTextView.context.getString(R.string.account_details_fee)}" }
 
                 if (ta.transactionStatus == TransactionStatus.RECEIVED ||
                     (ta.transactionStatus == TransactionStatus.COMMITTED && ta.outcome == TransactionOutcome.Ambiguous)
                 ) {
-                    amountTextView.setTextColor(colorBlack)
-                    costTextView.setTextColor(colorBlue)
                     costPrefix = "~"
+                    amountColorSpan = SpannableString(amountText).apply {
+                        setSpan(
+                            ForegroundColorSpan(colorBlack),
+                            0,
+                            amountText.length,
+                            0
+                        )
+                    }
+                    costColorSpan = SpannableString(costText).apply {
+                        setSpan(
+                            ForegroundColorSpan(colorBlue),
+                            0,
+                            costText.length,
+                            0
+                        )
+                    }
+                    textBuilder.append(amountColorSpan)
+                    textBuilder.append(costColorSpan)
```

Comment:
Would it make sense to have a local variable for the colours, and only set the spans/append text once after the conditional? 
(It seems to me that you are doing _almost_ the same thing in each branch.